onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -25,6 +31,21 @@ constexpr bool Is4BitIntType(int32_t data_type) {
          (data_type == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_UINT4);
 }
 
+FORCEINLINE bool IsPowerOfTwo(int64_t val) {
+  if (val < 0) return false;
+
+  bool seen_one = val & 1;
```

Comment:
`std::has_single_bit` once we have C++20...

https://en.cppreference.com/w/cpp/numeric/has_single_bit

here's a version for int64_t of their concise reference implementation:

`return val > 0 && !(val & (val - 1));`

Reply:
fix in next iter